monai/transforms/spatial/dictionary.py

```diff
@@ -286,11 +315,31 @@ def __init__(
     def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
         d = dict(data)
         for key in self.key_iterator(d):
+            self.push_transform(d, key)
             d[key] = self.rotator(d[key])
         return d
 
+    def inverse(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+        d = deepcopy(dict(data))
+        for key in self.key_iterator(d):
+            _ = self.get_most_recent_transform(d, key)
+            # Create inverse transform
+            spatial_axes = self.rotator.spatial_axes
```

Comment:
if we always use the undo stack to store `spatial_axes` and `k` instead of `self.rotator`, there's no need to  call `check_transforms_match` for the instance id. perhaps it makes the entire impl. more robust?

Reply:
Hm, I think the check still needs to happen. Otherwise a different transformation would be popped. Also, imagine `Rotate90d` followed by a crop. If the two are done in the wrong order, the resulting image would be the incorrect size.